app/controllers/spree/api/v1/user_sessions_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Spree
+  module Api
+    module V1
+      class UserSessionsController < Devise::SessionsController
+        skip_before_action :verify_authenticity_token
+
+        respond_to :json
+
+        def create
+          authenticate_spree_user!
+
+          if spree_user_signed_in?
+            respond_to do |format|
+              format.html {
+                flash[:success] = Spree.t(:logged_in_succesfully)
+                redirect_back_or_default(after_sign_in_path_for(spree_current_user))
+              }
+              format.js {
+                render json: { user: spree_current_user,
+                               ship_address: spree_current_user.ship_address,
+                               bill_address: spree_current_user.bill_address }.to_json
+              }
+              format.json {
+                render json: { success: true,
+                               auth_token: spree_current_user.authentication_token,
+                               spree_api_key: spree_current_user.spree_api_key,
+                               email: spree_current_user.email }
+              }
+            end
+          else
+            respond_to do |format|
+              format.html {
+                flash.now[:error] = t('devise.failure.invalid')
+                render :new
+              }
+              format.js {
+                render :json => { error: t('devise.failure.invalid') }, status: :unprocessable_entity
+              }
+              format.json {
+                render json: { success: false, error: t('devise.failure.invalid') }
```

Comment:
Same here, we need to use http error codes, this should return either forbidden or unprocessable_entity